test/BridgeProxy.test.js

```diff
@@ -0,0 +1,473 @@
+        token: genericErc20.address,
+        sender: ethers.encodeBytes32String("senderAddress"),
+        recipient: genericErc20.address,
+        amount: 80n,
+        depositNonce: 2,
+        callData: generateCallData(badFunctionSignature, gasLimit, argTypes, argValues),
+        isScRecipient: true,
+      };
+      arrayOfTxn.push(mvxTxnWithInvalidCalldata);
+      await prepareAndExecuteTransfer(amount, batchNonce, arrayOfTxn);
+
+      const beforeBalance = await genericErc20.balanceOf(bridge.address);
+      await bridgeProxy.execute(0);
+      const afterBalance = await genericErc20.balanceOf(bridge.address);
+
+      expect(afterBalance).to.equal(beforeBalance + mvxTxnWithInvalidCalldata.amount);
+
+      // check for pending transaction to be removed after refund
+      const pendingTxn = await bridgeProxy.getPendingTransactionById(0);
+      checkForEmptyTransaction(pendingTxn);
+    });
+
+    it("should gracefully finish execution and refund if calldata exists but endpoint is empty or gas limit is invalid (zero or below minimum)", async function () {
+      const functionSignature = "mint(address,uint256)";
+      const argTypes = ["address", "uint256"];
+      const argValues = [otherWallet.address, 999];
+
+      const mvxTxnWithZeroGas = {
+        token: genericErc20.address,
+        sender: ethers.encodeBytes32String("senderAddress"),
+        recipient: genericErc20.address,
+        amount: 80n,
+        depositNonce: 2,
+        callData: generateCallData(functionSignature, 0, argTypes, argValues),
+        isScRecipient: true,
+      };
+
+      const mvxTxnWithInsufficientGas = {
+        token: genericErc20.address,
+        sender: ethers.encodeBytes32String("senderAddress"),
+        recipient: genericErc20.address,
+        amount: 80n,
+        depositNonce: 2,
+        callData: generateCallData(functionSignature, 2000000, argTypes, argValues),
+        isScRecipient: true,
+      };
+
+      const mvxTxnWithEmptyEndpoint = {
+        token: genericErc20.address,
+        sender: ethers.encodeBytes32String("senderAddress"),
+        recipient: genericErc20.address,
+        amount: 80n,
+        depositNonce: 2,
+        callData: generateCallData("", 11000000, argTypes, argValues),
+        isScRecipient: true,
+      };
+
+      arrayOfTxn = [mvxTxnWithEmptyEndpoint, mvxTxnWithZeroGas, mvxTxnWithInsufficientGas];
+      await prepareAndExecuteTransfer(240, batchNonce, arrayOfTxn);
+
+      await bridgeProxy.execute(0);
+      await bridgeProxy.execute(1);
+      await bridgeProxy.execute(2);
+
+      // check for pending transaction to be removed after refund
+      checkForEmptyTransaction(bridgeProxy.getPendingTransactionById(0));
+      checkForEmptyTransaction(bridgeProxy.getPendingTransactionById(1));
+      checkForEmptyTransaction(bridgeProxy.getPendingTransactionById(2));
+    });
+
+    it("should successfully execute transaction with proper encoded calldata (with args)", async function () {
```

Comment:
After the allowance fix, we should also have a test with a contract that actually takes his funds from the bridge proxy contract. eg: after doing `await bridgeProxy.execute(0);`, `await genericErc20.balanceOf(bridgeProxy.address);` should be 0